app/assets/javascripts/i18n.js

```diff
@@ -497,6 +510,63 @@
     return message.replace(/_#\$#_/g, "$");
   };
 
+  // Not part of the public API
+  I18n.splitInterpolate = function(message, matches, options) {
+    var value
+    , placeholder
+    , name
+    , regex;
+
+    var buffer = [message];
+
+    while (matches.length) {
+      placeholder = matches.shift();
+      name = placeholder.replace(this.placeholder, "$1");
+
+      if (this.isSet(options[name])) {
+        value = options[name];
+      } else if (name in options) {
```

Comment:
For dynamic keys, you probably don't want to use the `in` operator to avoid checking properties in the prototype chain:

```javascript
> const o = { key: 123 }
> 'key' in o
true
> o.hasOwnProperty('key')
true
> 'toString' in o
true
> o.hasOwnProperty('toString')
false
```